### ISSUE TITLE
Warn about re-declaration of local variables

### ISSUE THREAD

**Issue by stefanheule** [microsoft/TypeScript#656]
TypeScript (tsc to be precise) should warn about colliding local variable declarations (two declarations with the same name, in the same scope).  For instance, the following (TypeScript) program should cause a warning:

``` javascript
var test = 0;
var test = 1;
```

Since JavaScript (and thus TypeScript) don't introduce a new scope for brackets {}, this can lead to subtle problems that could be easily caught by a compiler warning about re-declarations.  Consider the following code:

``` javascript
for (var i = 0; i < 10; i++) {
  // ... (other code)
  for (var i = 2; i < 4; i++) {
    console.log("test");
  }
}
```

It's easy to overlook that `i` refers to the same variable in both loops.
